File: problems/929-unique-email-addresses/solution.py
class Solution:
    def numUniqueEmails(self, emails: list[str]) -> int:
        unique_emails = set()
        for email in emails:
            local_name, domain_name = email.split('@')
            local_name = local_name.split('+')[0]
            local_name = local_name.replace('.', '')
            unique_emails.add(local_name + '@' + domain_name)
        return len(unique_emails)

File: problems/929-unique-email-addresses/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int numUniqueEmails(String[] emails) {
        Set<String> uniqueEmails = new HashSet<>();
        for (String email : emails) {
            String[] parts = email.split("@");
            String localName = parts[0];
            String domainName = parts[1];
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < localName.length(); i++) {
                char c = localName.charAt(i);
                if (c == '+') {
                    break;
                }
                if (c != '.') {
                    sb.append(c);
                }
            }
            uniqueEmails.add(sb.toString() + "@" + domainName);
        }
        return uniqueEmails.size();
    }
}